src/common/config.cc

```diff
@@ -1176,12 +1201,29 @@ void Config::GenData() {
   ul_iq_ifft.Calloc(this->frame_.NumULSyms(),
                     this->ofdm_ca_num_ * this->ue_ant_num_,
                     Agora_memory::Alignment_t::kAlign64);
+  std::vector<FILE*> vec_fp_tx;
+  if (kOutputUlFreqData) {
+    for (size_t u = 0; u < this->ue_ant_num_; u++) {
+      const std::string filename_ul_data_f =
+          kUlDataFreqPrefix + ul_modulation_ + "_" +
+          std::to_string(ofdm_data_num_) + "_" + std::to_string(ofdm_ca_num_) +
+          "_1_" + std::to_string(this->frame_.NumULSyms()) + "_1_A_" +
```

Comment:
This was a hack I did originally to make this work with Sounder. You see some of the file name parts are not paramertized, e.g. `_1_` which I think refers to the number of UEs, and `_1_A_` which refer to channels used. We should parameterize them and also only create the file when testing with Sounder.